src/contexts/payment-context.tsx

```diff
@@ -50,29 +54,41 @@ export const PaymentContextProvider = ({
     React.useState<PaymentTier | null>(null);
   const [isAutoPaymentSettled, setIsAutoPaymentSettled] = React.useState(false);
   const [paymentCancelled, setPaymentCancelled] = React.useState(false);
+  const [preferAutoPayment, setPreferAutoPayment] = useLocalStorage(
+    "prefer-auto-payment",
+    false
+  );
 
   const openPaymentModal = React.useCallback(() => {
     setIsPaymentModalOpen(true);
   }, []);
 
-  const requestAutoPayment = React.useCallback(async (priceInSats: number) => {
-    setIsAutoPaymentSettled(false);
-    setAutoPaymentInvoice(defaultInvoice);
-    setAutoPaymentLoading(true);
-    const response = await fetch("/api/server", {
-      method: "POST",
-      body: JSON.stringify({ autoPayment: priceInSats }),
-    });
-    if (response.status === 402) {
-      window.localStorage.removeItem("paymentToken");
-      const L402 = response.headers.get("WWW-Authenticate");
-      const { token, invoice, r_hash } = getLSATDetailsFromHeader(L402!) ?? {};
-      localStorage.setItem("paymentToken", token ?? "");
-      setAutoPaymentInvoice({ payment_request: invoice!, r_hash: r_hash! });
-      setAutoPaymentLoading(false);
-      return { payment_request: invoice!, r_hash: r_hash! };
-    }
-  }, []);
+  const autoPayWithWebln = React.useCallback(
+    async (invoice?: string) => {
+      const payment_request = invoice ?? autoPaymentInvoice.payment_request;
+      try {
+        const webln = await requestProvider();
+        if (!webln) {
+          setError("webln not available");
+          setIsPaymentModalOpen(true);
+          return { error: null, preimage: null };
+        }
+
+        const res = await webln.sendPayment(payment_request);
+        if (res instanceof Error) {
+          setError("could not pay with webln");
+          setIsPaymentModalOpen(true);
+          return { error: res, preimage: null };
+        }
+
+        return { error: null, preimage: res.preimage };
+      } catch (error) {
+        setError("could not pay with webln");
```

Comment:
any reason why we are not showing the payment modal like we are on `line 80`

Reply:
Its an over-sight. will add it

Reply:
fixed here -> https://github.com/bitcoinsearch/chat-btc/pull/52/commits/7b3154b304cbf08c03af22ee048c9ab69ba50c30